### DISCUSSION TITLE
Move nested Entry from one field to another

### DISCUSSION THREAD

**Discussion by shifuma** [craftcms/cms#15776] [Q&A] [answered]
How portable are nested entries? I have this setup:

```
- Entry
- - Nested Entry
- - - Entry to Move
```

I would like to update it to:

```
- Entry
- - Nested Entry
- - Entry to Move 
```

So basically change the owner of `Entry to Move`. I'm using the same Matrix Field and the same Entry Type.

I see `primaryOwnerId` is stored in the database, is there a way I can update that with Craft APIs? Or would I have to create new entries and delete the old ones?

Edit: I tried this, but to no avail:

```
$owner = $nestedEntry->getRootOwner();
$nestedEntry->setOwner($owner);
Craft::$app->elements->saveElement($nestedEntry);
```

**Comment by thupsi** [comment]
Not sure if there's already a FR for this, but it would be very useful if authors were given the ability to move nested entries to not only other fields, but also directly to sections (un-nest them). And vice versa: Move regular entries to a matrix field. I haven't a clear idea how that would look UI wise, but for starters it could be a three-step thing, like select `move to field or section` on an entry's menu, then navigate to wherever, and finaly select `move selected entry here`. Or something along those lines.

**Comment by brandonkelly** [comment, marked as answer]
Thanks for posting! This was previously requested via #14469.

**Reply by shifuma** [reply]
Thanks for the reply. I did see that, but it seems to be specifically asking for a UI to do it, but I need to do it on 100 entries or so so was looking to do it programmatically. Maybe what you're saying is that when the UI is available for it, it will also be available via code...but I just wanted to confirm in case the above reply confused what I was asking for.

**Reply by brandonkelly** [reply]
Gotcha. There’s no API-based way to do it currently. You would need to manually update the nested entry’s `primaryOwnerId` value in the `entries` table, as well as the `ownerId` value for any rows in the `elements_owners` table. You’d also need to look for any drafts in the `drafts` table where the `canonicalId` value points to the nested entries, and update their `primaryOwnerId` + `ownerId` values as well.
